Scenarios/CGCT/CGCT/ViewController.m

```diff
@@ -74,28 +75,32 @@ - (void)viewDidLoad {
 
     [self.view addSubview:_stage];
     [self.view addSubview:_CGMenu];
+
+	_demoList = [[NSMutableArray alloc] init];
+	[_demoList addObject:[[DemoScenarioIntro alloc] init]];
 }
 
 - (void)viewWillAppear:(BOOL)animated {
 }
 
 - (void)tableView:(UITableView*)tableView didSelectRowAtIndexPath:(NSIndexPath*)indexPath {
     if (tableView == _CGMenu) {
-        _stage.backgroundColor = [UIColor colorWithRed:.1 green:.3 blue:1 alpha:1];
-        [_stage updateCurrentDemo:[tableView cellForRowAtIndexPath:indexPath]];
-        [_stage setNeedsDisplay];
+		_stage.backgroundColor = [UIColor colorWithRed : .1 green : .3 blue : 1 alpha : 1];
+		[_stage updateCurrentDemo:[_demoList objectAtIndex:indexPath.row]];
+        
+		[_stage setNeedsDisplay];
     }
 }
 
 - (NSInteger)tableView:(UITableView*)tableView numberOfRowsInSection:(NSInteger)section {
-    return 1;
+    return _demoList.count;
 }
 
 - (UITableViewCell*)tableView:(UITableView*)tableView cellForRowAtIndexPath:(nonnull NSIndexPath*)indexPath {
     if (tableView == _CGMenu) {
         UITableViewCell* aCell = [tableView dequeueReusableCellWithIdentifier:@"CGStory1"];
         if (aCell == nil) {
-            aCell = [[PathDemoTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"CGStory1"];
+            aCell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"CGStory1"];
         }
         aCell.backgroundColor = [UIColor colorWithRed:1 green:1 blue:1 alpha:1];
         aCell.textLabel.text = @"CoreDemo1";
```

Comment:
this could read `demo.name` where `demo` comes from the list.

also demos would need names.